executor/src/test/java/org/k8loud/executor/service/MapperServiceImplTest.java

```diff
@@ -66,24 +45,23 @@ void testValidMapping(String collectionName, String actionName, Map<String, Stri
 
     @ParameterizedTest
     @MethodSource("provideUnmappableExecutionRQs")
-    void testInvalidMapping(String collectionName, String actionName, Map<String, String> params, MapperExceptionCode exceptionCode) {
+    void testInvalidMapping(String collectionName, String actionName, Map<String, String> params,
+                            MapperExceptionCode expectedExceptionCode) {
         // given
         ExecutionRQ executionRQ = createExecutionRQ(collectionName, actionName, params);
 
-        try {
-            // when
-            mapperService.map(executionRQ);
-            fail();
-        } catch (MapperException e) {
-            // then
-            Assertions.assertEquals(e.getExceptionCode(), exceptionCode);
-        }
+        // when
+        Throwable e = catchThrowable(() -> mapperService.map(executionRQ));
+
+        // then
+        Assertions.assertEquals(MapperException.class, e.getClass());
```

Comment:
and what about `assertThat(e).isExaclyInstanceOf(MapperException.class)`? Also (probably) might be added `.hasExceptionCode(e.getExceptionCode())`